tests/integration/backup/s3/test_backup.py

```diff
@@ -130,279 +131,510 @@ def test__take_mysql_backup(master1,
+
+    cmd = ['twindb-backup', '--debug', '--config', twindb_config_guest, 'backup', 'hourly']
+    for i in range(0, 3):
+        ret, cout = docker_execute(docker_client, master1['Id'], cmd)
+        print(cout)
+        assert ret == 0
+
+    cmd = ['twindb-backup', '--config', twindb_config_guest, 'status']
+    ret, cout = docker_execute(docker_client, master1['Id'], cmd)
+    assert ret == 0
+    print(cout)
+    status = json.loads(cout)
+
+    assert len(status['daily'].keys()) == 1
+    assert len(status['hourly'].keys()) == 2
+
+
+def test__s3_find_files_returns_sorted(master1,
+                                       docker_client,
+                                       s3_client,
+                                       config_content_mysql_only):
+    # cleanup the bucket first
+    s3_client.delete_all_objects()
+
+    twindb_config_dir = get_twindb_config_dir(docker_client, master1['Id'])
+
+    twindb_config_host = "%s/twindb-backup-1.cfg" % twindb_config_dir
+    twindb_config_guest = '/etc/twindb/twindb-backup-1.cfg'
+    my_cnf_path = "%s/my.cnf" % twindb_config_dir
+
+    contents = """
+[client]
+user=dba
+password=qwerty
+"""
+
+    with open(my_cnf_path, "w") as my_cnf:
+        my_cnf.write(contents)
+
+    with open(twindb_config_host, 'w') as fp:
+        content = config_content_mysql_only.format(
+            AWS_ACCESS_KEY_ID=os.environ['AWS_ACCESS_KEY_ID'],
+            AWS_SECRET_ACCESS_KEY=os.environ['AWS_SECRET_ACCESS_KEY'],
+            BUCKET=s3_client.bucket,
+            daily_copies=5,
+            hourly_copies=2,
+            MY_CNF='/etc/twindb/my.cnf'
+        )
+        fp.write(content)
+
+    cmd = ['twindb-backup', '--debug', '--config', twindb_config_guest, 'backup', 'daily']
+
+    n_runs = 3
+    for x in xrange(n_runs):
+        ret, cout = docker_execute(docker_client, master1['Id'], cmd)
+        print(cout)
+        assert ret == 0
+    hostname = 'master1_1'
+    dst = S3(s3_client.bucket,
+             AWSAuthOptions(os.environ['AWS_ACCESS_KEY_ID'],
+                            os.environ['AWS_SECRET_ACCESS_KEY'])
+             )
+
+    for x in xrange(10):
+        result = dst.find_files(dst.remote_path, 'daily')
+        assert len(result) == n_runs
+        assert result == sorted(result)
+        prefix = "{remote_path}/{hostname}/{run_type}/mysql/mysql-".format(
+            remote_path=dst.remote_path,
+            hostname=hostname,
+            run_type='daily'
+        )
+        files = dst.list_files(prefix)
+        assert len(files) == n_runs
+        assert files == sorted(files)
+
+
+# def test__take_mysql_backup_aenc_suffix_gpg(master1,
+#                                             docker_client,
+#                                             s3_client,
+#                                             config_content_mysql_aenc,
+#                                             gpg_keyring,
+#                                             gpg_secret_keyring):
 #
```

Comment:
The change looks good, but a commented out part polluted the PR.
Please revert changes in the commented out tests.